homeassistant/components/homekit/aidmanager.py

```diff
@@ -0,0 +1,125 @@
+"""
+Manage allocation of accessory ID's.
+
+HomeKit needs to allocate unique numbers to each accessory. These need to
+be stable between reboots and upgrades.
+
+Using a hash function to generate them means collisions. It also means you
+can't change the hash without causing breakages for HA users.
+
+This module generates and stores them in a HA storage.
+"""
+import hashlib
+import logging
+import random
+from zlib import adler32
+
+from homeassistant.core import callback
+from homeassistant.helpers.storage import Store
+
+from .const import DOMAIN
+
+AID_MANAGER_STORAGE_KEY = f"{DOMAIN}-aid-storage"
+AID_MANAGER_STORAGE_VERSION = 1
+AID_MANAGER_SAVE_DELAY = 2
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def generate_aids(unique_id, entity_id):
+    """Generate accessory aid with zlib adler32."""
+
+    # Backward compatibility: Previously HA used to *only* do adler32 on the entity id.
+    # Not stable if entity ID changes
+    # Not robust against collisions
+    yield adler32(entity_id.encode("utf-8"))
+
+    # Use adler32 on a sha of the unique id
+    yield adler32(hashlib.sha512(unique_id.encode("utf-8")).digest())
```

Comment:
@Jc2k Since unique_ids are per domain should the DOMAIN be included in the hash?

Reply:
We only enforce uniqueness among the combination of entity domain + entity platform + unique_id, so you need all 3.

Reply:
Ah, face palm, of course. Thank for reminding me! I'll try and rework this in after the Ecobee Switch fix i'm working on.

Reply:
@Jc2k How about using `fnv1a` (maybe https://pypi.org/project/fnvhash/) or another non-cryptographically secure hash that is less collision prone than `adler32`?

Reply:
Actually why not just drop the second hashing and just pick a random number since you this is going to be back compat anyways and its only ever an issue if they loose the state file

Reply:
Right after I wrote that, I realized, zero chance someone isn't going to loose the state file or delete it. 
So maybe `fnv1a` does make sense here.

Reply:
If we  store the state file in the config dir, they loose a lot more if they reset